src/main/java/mezz/jei/render/ItemStackFastRenderer.java

```diff
@@ -49,65 +43,27 @@ private IBakedModel getBakedModel() {
 	private void uncheckedRenderItemAndEffectIntoGUI(IEditModeConfig editModeConfig, IWorldConfig worldConfig) {
 		if (worldConfig.isEditModeEnabled()) {
 			renderEditMode(area, padding, editModeConfig);
-			GlStateManager.enableBlend();
+			RenderSystem.enableBlend();
 		}
 
 		ItemStack itemStack = element.getIngredient();
 		IBakedModel bakedModel = getBakedModel();
 		if (bakedModel == null) {
 			return;
 		}
-
-		GlStateManager.pushMatrix();
+		RenderSystem.pushMatrix();
 		{
-			GlStateManager.translatef(area.getX() + padding + 8.0f, area.getY() + padding + 8.0f, 150.0F);
-			GlStateManager.scalef(16F, -16F, 16F);
-			bakedModel = ForgeHooksClient.handleCameraTransforms(bakedModel, ItemCameraTransforms.TransformType.GUI, false);
-			GlStateManager.translatef(-0.5F, -0.5F, -0.5F);
-
+			RenderSystem.translatef(area.getX() + padding + 16.0F, area.getY() + padding, 150.0F);
+			RenderSystem.scalef(16F, -16F, 16F);
+			RenderSystem.translatef(-0.5F, -0.5F, -0.5F);
+			MatrixStack matrixStack = new MatrixStack();
```

Comment:
Not sure given not bothering to download the branch to experiment, but would it make more sense/be more performant to make the tralsations/scales directly to the matrix stack, rather then doing them via OpenGL?
```java
MatrixStack matrixStack = new MatrixStack();
matrixStack.func_227861_a_(area.getX() + padding + 16, area.getY() + padding, 150);
matrixStack.func_227862_a_(16, -16, 16);
matrixStack.func_227861_a_(-0.5, -0.5, -0.5);
```
(Note: func_227861_a_ which is the translate for a matrix stack, takes doubles instead of floats)

Also this way I believe it would have less GL calls, and you could remove the RenderSystem.push/pop matrix (I didn't bother adding the push/pop for matrix stack as it is a new matrix stack you are using that then gets GC'd after it is done being used). Also you can probably remove the `{`, `}` that are between the push/pop regardless.

Reply:
Yeah I did test that and would probably improve it beyond what it was in previous versions, was just getting the same functionality and looking at how it was done in vanilla, plus trying to keep it readable, but I've changed it now.

Reply:
Ya I figured that it would work just wanted to make sure to mention it so that you would test it before replacing. Also figured you probably had not looked at translation stuff directly on the matrix stack due to it being a simple port, so figured I would mention it given I have lately been messing around with converting another mod from using GL calls in TESRs and the like to modifying the matrix stack instead.

Reply:
Thanks for the suggestion, I have been building up quite the dictionary of function names when porting some of my other TESRs as well, so will be ready to feed them in once we have a working MCP replacement. With 1.15 being new, going to take a bit of time to get used to new ways of optimizing OpenGL Calls.